addons/addon-base-raas/packages/base-raas-services/lib/data-source/data-source-reachability-service.js

```diff
@@ -0,0 +1,197 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+// const _ = require('lodash');
+const Service = require('@aws-ee/base-services-container/lib/service');
+const { allowIfActive, allowIfAdmin } = require('@aws-ee/base-services/lib/authorization/authorization-utils');
+
+const workflowIds = {
+  bulkCheck: 'wf-bulk-reachability-check',
+  accountStatusChange: 'wf-ds-account-status-change',
+};
+
+class DataSourceReachabilityService extends Service {
+  constructor() {
+    super();
+    this.dependency([
+      'auditWriterService',
+      'dataSourceAccountService',
+      'studyService',
+      'workflowTriggerService',
+      'aws',
+    ]);
+  }
+
+  async bulkReach(requestContext, { status }, forceCheckAll = false) {
```

Comment:
What do you think about making third argument as an option object? This will allow us to add more options in the future. So something like this:
`async bulkReach(requestContext, { status }, { forceCheckAll = false } = {}) `

Reply:
Done